CMakeLists.txt

```diff
@@ -13,236 +13,280 @@
-	message(STATUS "Using Boost_USE_RELEASE_LIBS: ${Boost_USE_RELEASE_LIBS}")
-endif()
-find_package(Boost REQUIRED COMPONENTS thread chrono)
-include_directories(${Boost_INCLUDE_DIRS})
-link_libraries(Boost::thread Boost::chrono)
-add_definitions("-DBOOST_THREAD_VERSION=5")
-message(STATUS "Using Boost_INCLUDE_DIRS: ${Boost_INCLUDE_DIRS}")
-message(STATUS "Using Boost_LIBRARY_DIRS: ${Boost_LIBRARY_DIRS}")
-
-IF (NOT (${HZ_BIT} MATCHES "32") AND NOT (${HZ_BIT} MATCHES "64"))
-    message(STATUS "Build needs HZ_BIT. Setting default as -DHZ_BIT=64 (other option -DHZ_BIT=32)")
-    set(HZ_BIT 64)
-ENDIF (NOT (${HZ_BIT} MATCHES "32") AND NOT (${HZ_BIT} MATCHES "64"))
-
-IF (NOT (${HZ_LIB_TYPE} MATCHES "STATIC") AND NOT (${HZ_LIB_TYPE} MATCHES "SHARED"))
-	message(STATUS "Build needs HZ_LIB_TYPE. Setting default as -DHZ_LIB_TYPE=STATIC (other option -DHZ_LIB_TYPE=SHARED)")
-	set(HZ_LIB_TYPE STATIC)
-ENDIF(NOT (${HZ_LIB_TYPE} MATCHES "STATIC") AND NOT (${HZ_LIB_TYPE} MATCHES "SHARED") )
-
-message(STATUS "Generating ${HZ_LIB_TYPE} Library")
-
-SET(HZ_LIB_NAME HazelcastClient${HZ_VERSION}_${HZ_BIT})
-
-set(HZ_CODE_COVERAGE_COMPILE_FLAGS " ")
-set(HZ_CODE_COVERAGE_LINK_FLAGS " ")
 
-IF(HZ_CODE_COVERAGE)
-	message(STATUS "Code coverage is on")
-	set(HZ_CODE_COVERAGE_COMPILE_FLAGS " -fprofile-arcs -ftest-coverage -fPIC -O0 ")
-	set(HZ_CODE_COVERAGE_LINK_FLAGS " -lgcov -fprofile-arcs -ftest-coverage -fPIC -O0 ")
-ENDIF(HZ_CODE_COVERAGE)
-
-set(HZ_VALGRIND_COMPILE_FLAGS " ")
-set(HZ_VALGRIND_LINK_FLAGS " ")
-IF(HZ_VALGRIND)
-	message(STATUS "Valgrind is on")
-	set(HZ_VALGRIND_COMPILE_FLAGS " -O0 ")
-	set(HZ_VALGRIND_LINK_FLAGS " -O0 ")
-ENDIF(HZ_VALGRIND)
-
-message(STATUS "${CMAKE_SYSTEM}")
-IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-	message(STATUS "APPLE ENVIRONMENT DETECTED")
-
-	set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -Wall -Werror -fsanitize=address -fno-omit-frame-pointer")
-	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -Wall")
-
-	add_library(${HZ_LIB_NAME} ${HZ_LIB_TYPE} ${HZ_SOURCES} ${INLINE_SOURCES} ${HZ_GENERATED_SOURCES} ${HZ_HEADERS} ${HZ_GENERATED_HEADERS})
-
-	link_libraries(${HZ_LIB_NAME})
-
-ENDIF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+include(FindThreads)
+include(FindBoost)
+include(FindOpenSSL)
+include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
+include(./cmake/utils.cmake)
+
+# get and set Git commit date and id
+get_commit_date_and_id(GIT_COMMIT_DATE GIT_COMMIT_ID)
+message(STATUS "GIT_COMMIT_DATE = ${GIT_COMMIT_DATE}")
+message(STATUS "GIT_COMMIT_ID = ${GIT_COMMIT_ID}")
+
+# find all source and header files
+FILE(GLOB_RECURSE SOURCE_FILES
+     "./hazelcast/src/*cpp"
+     "./hazelcast/generated-sources/src/*cpp")
+FILE(GLOB_RECURSE HEADER_FILES
+     "./hazelcast/include/*h"
+     "./hazelcast/generated-sources/src/*h")
+
+# options
+option(WITH_OPENSSL
+       "Use OpenSSL, setting this option to OFF causes SSL-related features to be disabled."
+       ON)
```

Comment:
should we have the default version SSL OFF not to cause OpenSSL dependency by default?

Reply:
Yes, you are right, we should. I will change the default to `OFF`.